authentication/private_key_signer.go

```diff
@@ -20,12 +20,13 @@ type PrivateKeySigner struct {
 	keyFingerprint          string
 	algorithm               string
 	accountName             string
+	userName                string
 	hashFunc                crypto.Hash
 
 	privateKey *rsa.PrivateKey
 }
 
-func NewPrivateKeySigner(keyFingerprint string, privateKeyMaterial []byte, accountName string) (*PrivateKeySigner, error) {
+func NewPrivateKeySigner(keyFingerprint string, privateKeyMaterial []byte, accountName, userName string) (*PrivateKeySigner, error) {
```

Comment:
Sweet, this is a good breaking change.  Changelog++  I'm assuming you, @stack72 , are going to chase the common breakage in tooling? https://godoc.org/github.com/joyent/triton-go?importers

btw, the reason I'm okay with this API change is because the API should specify the username that has signed the request.  Implicit identity is "problematic."

I'd rather suck up the explicit breakage now vs introduce something using an ellipsis that we later have to rip out.

Reply:
... you know, since we're making a breaking change, do we want to just outright break it and use an input struct?

```go
type NewPrivateKeySignerInput struct {
  KeyFingerPrint string
  PrivateKeyMaterial []byte
  AccountName string
  Username string
}
```

And pass that in (by value, ideally)?

This would make the API future-proof from additions like delegation. ?

Reply:
I really like this idea! I will change to that

Reply:
:+1:, this matches the rest of the API at that point too.